packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -52,39 +52,49 @@ export const createCart = payload =>
     };
 
 export const addItemToCart = (payload = {}) => {
-    const { item, fetchCartId } = payload;
+    const {
+        addConfigurableProductToCart,
+        addSimpleProductToCart,
+        fetchCartId,
+        item,
+        productType,
+        quantity,
+        parentSku
+    } = payload;
+
     const writingImageToCache = writeImageToCache(item);
 
     return async function thunk(dispatch, getState) {
         await writingImageToCache;
         dispatch(actions.addItem.request(payload));
 
         try {
-            const { cart, user } = getState();
-            const { isSignedIn } = user;
-            let cartEndpoint;
-
-            if (!isSignedIn) {
-                const { cartId } = cart;
-
-                if (!cartId) {
-                    const missingCartIdError = new Error(
-                        'Missing required information: cartId'
-                    );
-                    missingCartIdError.noCartId = true;
-                    throw missingCartIdError;
-                }
+            const { cart } = getState();
+            const { cartId } = cart;
 
-                cartEndpoint = `/rest/V1/guest-carts/${cartId}/items`;
+            const variables = {
+                cartId,
+                parentSku,
+                product: item,
+                quantity,
+                sku: item.sku
+            };
+
+            let addItemMutation;
+
+            // Prepare the params to add and use the proper mutation.
+            if (productType === 'SimpleProduct') {
```

Comment:
This might prove to be a terrible idea once we add other product types, but hear me out. If we move this logic to `useProductFullDetail`, you can pass a single addItem mutation to the action. I hope we don't sit in this intermediate phase too long, but limiting action dependencies seems preferable here.

Whatchu think?

Reply:
> If we move this logic to useProductFullDetail, you can pass a single addItem mutation to the action.

I had it like that originally but I was also doing logic to create the `variables` object within the talons rather than the action. I'll see what this looks like in the talons. Remember, `updateItemInCart` will need to behave the same. I'll make the change there as well if it looks OK here.

> limiting action dependencies seems preferable here.

I agree.  I think it may make refactoring easier later.